lib/SIL/SILFunctionBuilder.cpp

```diff
@@ -162,8 +162,17 @@ SILFunctionBuilder::getOrCreateFunction(SILLocation loc, SILDeclRef constant,
     if (constant.isForeign && decl->hasClangNode())
       F->setClangNodeOwner(decl);
 
-    if (decl->isWeakImported(/*forModule=*/nullptr, availCtx))
+    if (decl->isWeakImported())
       F->setWeakLinked();
+    else {
+      auto containingContext = decl->getAvailabilityForLinkage();
+      if (!containingContext.isAlwaysAvailable()) {
+        auto fromContext = AvailabilityContext::forDeploymentTarget(
+          decl->getASTContext());
+        if (!fromContext.isContainedIn(containingContext))
+          F->setWeakLinked();
```

Comment:
What's the difference between this whole block and `isWeakImported(/*fromModule*/nullptr)`? The fact that it's not obvious to me makes me concerned about misuse.

Reply:
Take a look at the next patch, this gets cleaned up again. Its just there to keep tests passing in the half-refactored state

Reply:
Oh, GitHub reordered them again…